front/app/containers/Authentication/useSteps/index.ts

```diff
@@ -289,7 +304,7 @@ export default function useSteps() {
 
       if (sso_pathname) {
         clHistory.replace(sso_pathname);
-      } else {
+      } else if (!verification_success) {
         // Remove all parameters from URL as they've already been captured
         window.history.replaceState(null, '', '/');
       }
```

Comment:
If we replaceState for cases with verification, it redirects the user to the homepage rather than where they were previously.